### ISSUE TITLE
CSV parsing utf-8 first field incorrectly

### ISSUE THREAD

**Issue by montaguethomas** [mikefarah/yq#1373]
**Describe the bug**
CSV files with UTF-8 byte order mark (`0xEF,0xBB,0xBF`) have the first field incorrectly parsed/rendered.

Normal ascii files work fine.

Version of yq: 4.27.5
Operating system: mac
Installed via: `brew install yq`

**Input CSV**
names.csv (in hex):
```hex
EFBBBF69 642C6669 7273742C 6C617374 0D0A312C 6A6F686E
2C736D69 74680D0A 312C6A61 6E652C73 6D697468 0D0A
```

**Command**
The command you ran:
```
yq -v -p csv names.csv
```

**Actual behavior**
```
$ yq -v -p csv names.csv
12:11:21 processArgs [DEBU] processed args: [names.csv]
12:11:21 maybeFile [DEBU] checking 'names.csv' is a file
12:11:21 maybeFile [DEBU] error: <nil>, dir: false
12:11:21 maybeFile [DEBU] result: true
12:11:21 ParseExpression [DEBU] Parsing expression: []
12:11:21 ConvertToPostfix [DEBU] postfix processing currentToken )
12:11:21 ConvertToPostfix [DEBU] opstackLen: 0
12:11:21 ConvertToPostfix [DEBU] PostFix Result:
12:11:21 EvaluateFiles [DEBU] leadingContent:
12:11:21 Decode [DEBU] : headerRow[id first last]
12:11:21 Decode [DEBU] Adding contentRow: [1 john smith]
12:11:21 Decode [DEBU] Read next contentRow: [1 jane smith], <nil>
12:11:21 Decode [DEBU] Adding contentRow: [1 jane smith]
12:11:21 Decode [DEBU] Read next contentRow: [], EOF
12:11:21 Decode [DEBU] finished, contentRow[]
12:11:21 Decode [DEBU] err: EOF
12:11:21 GetMatchingNodes [DEBU] getMatchingNodes - nothing to do
12:11:21 PrintResults [DEBU] PrintResults for 1 matches
12:11:21 PrintResults [DEBU] -- print sep logic: p.firstTimePrinting: false, previousDocIndex: 0, mappedDoc.Document: 0
- "\\\": 1
  first: john
  last: smith
- "\\\": 1
  first: jane
  last: smith
12:11:21 PrintResults [DEBU] done printing results
12:11:21 Decode [DEBU] : headerRow[]
```

**Expected behavior**

```
$ yq -p csv names.csv
- id: 1
  first: john
  last: smith
- id: 1
  first: jane
  last: smith
```

**Additional information**

```
$ file -I names-utf8.csv
names-utf8.csv: text/csv; charset=utf-8

$ file -I names-ascii.csv
names-ascii.csv: text/csv; charset=us-ascii
```

**Comment by mikefarah** [comment]
Thanks, will fix in the next release

**Comment by mikefarah** [comment]
Fixed in 4.28.2
